envoy/router/router.h

```diff
@@ -540,6 +540,11 @@ class ShadowPolicy {
    */
   virtual bool traceSampled() const PURE;
 
+  /**
+   * @return true if the config has `trace_sampled` set.
+   */
+  virtual bool hasTraceSampledConf() const PURE;
+
```

Comment:
Please just change the traceSampled() method to return an `absl::optional<bool>` to avoid adding new method.

Reply:
Done :)